templates/preview/previewApi.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/tidepool-org/go-common/clients/status"
+	"github.com/tidepool-org/hydrophone/localize"
+	"github.com/tidepool-org/hydrophone/models"
+	"github.com/tidepool-org/hydrophone/templates"
+)
+
+type (
+	Api struct {
+		Config        Config
+		templates     models.Templates
+		localeManager LocaleManager
+	}
+	// this just makes it easier to bind a handler for the Handle function
+	varsHandler func(http.ResponseWriter, *http.Request, map[string]string)
+)
+
+// Init the preview api with configuration
+func InitApi(
+	cfg Config,
+	templates models.Templates,
+	localeManager LocaleManager,
+) *Api {
+	return &Api{
+		Config:        cfg,
+		templates:     templates,
+		localeManager: localeManager,
+	}
+}
+
+func (a *Api) SetHandlers(prefix string, rtr *mux.Router) {
+	rtr.Handle("/preview/{template}", varsHandler(a.preview)).Methods("GET")
+	rtr.Handle("/refreshlocal", varsHandler(a.refreshLocal)).Methods("POST")
+	rtr.HandleFunc("/", a.serveStatic).Methods("GET")
+}
+
+func (h varsHandler) ServeHTTP(res http.ResponseWriter, req *http.Request) {
+	vars := mux.Vars(req)
+	h(res, req, vars)
+}
+
+// Return the index page
+func (a *Api) serveStatic(res http.ResponseWriter, req *http.Request) {
+	data, err := ioutil.ReadFile("index.html")
+	if err != nil {
+		log.Printf("templates - failure to read index.html")
+	}
+	res.Header().Set("content-type", "text/html")
+	res.WriteHeader(200)
+	res.Write(data)
+	return
+}
+
+// Refresh locales from the localization system (i.e. Loco)
+func (a *Api) refreshLocal(res http.ResponseWriter, req *http.Request, vars map[string]string) {
```

Comment:
If the refresh local failed, the web page is stuck. Can preview anything anymore.
The server need to be restarted.

Reply:
Curious. I don't observe this behaviour locally.
What do you suggest?